src/app/modules/form/admin-form/admin-form.service.ts

```diff
@@ -582,3 +586,52 @@ export const updateFormSettings = (
     return okAsync(updatedForm.getSettings())
   })
 }
+
+/**
+ * Deletes form logic.
+ * @param form The original form to delete logic in
+ * @param logicId the logicId to delete
+ * @returns ok(true) on success
+ * @returns err(database errors) if db error is thrown during logic delete
+ * @returns err(LogicNotFoundError) if logicId does not exist on form
+ */
+export const deleteFormLogic = (
+  form: IPopulatedForm,
+  logicId: string,
+): ResultAsync<IFormSchema, DatabaseError | LogicNotFoundError> => {
+  // First check if specified logic exists
+  if (!form.form_logics.some((logic) => logic.id === logicId)) {
+    logger.error({
+      message: 'Error occurred - logicId to be deleted does not exist',
+      meta: {
+        action: 'deleteFormLogic',
+        formId: form._id,
+        logicId,
+      },
+    })
+    return errAsync(new LogicNotFoundError())
+  }
+
+  // Remove specified logic and then update form logic
+  return ResultAsync.fromPromise(
+    FormModel.deleteFormLogic(form._id, logicId),
```

Comment:
sorry final nit - should we do `String(form._id)` instead? `._id` has a nasty habit of giving us an unexpected type (e.g. ObjectId) at runtime.

Reply:
in this case it does not matter since mongoose will cast it properly, but yea why not

Reply:
it'll cast properly when we call findByIdAndUpdate, but the typing is still wrong

Reply:
yea, better be safe than sorry!

Reply:
updated